prototype2/gdgem/clustering/test_data/OldReadoutFile.h

```diff
@@ -0,0 +1,86 @@
+/* Copyright (C) 2016-2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cinttypes>
+#include <string>
+#include <common/DumpFile.h>
+
+// \todo initialize values to 0?
+struct OldReadout
+{
+  uint8_t fec;
+  uint8_t chip_id;
+  uint32_t bonus_timestamp;
+  uint32_t srs_timestamp;
+  uint16_t channel;
+  uint16_t bcid;
+  uint16_t tdc;
+  uint16_t adc;
+  bool over_threshold;
+
+  // \todo use constexpr string_view when c++17 arrives
+  static std::string DatasetName() { return "srs_hits"; }
+  static std::string FormatVersion() { return "1.0.0"; }
+};
+
+
+namespace hdf5 {
+
+namespace datatype {
+template<>
+class TypeTrait<OldReadout>
+{
+private:
+  static constexpr size_t u8 { sizeof(std::uint8_t)};
+  static constexpr size_t u16 { sizeof(std::uint16_t)};
+  static constexpr size_t u32 { sizeof(std::uint32_t)};
+  static constexpr size_t b { sizeof(bool)};
+
+public:
+  using Type = OldReadout;
+  using TypeClass = Compound;
+
+  static TypeClass create(const Type & = Type())
+  {
+    auto type = datatype::Compound::create(sizeof(OldReadout));
```

Comment:
Is there a good reason to use the "old" compound data creation style? Can we not replace it for the new and safer version?